SpectodaParser.js

```
@@ -482,43 +612,79 @@
 
     const const_name = reg[1];
-    const const_address = CONST_VALUE_ADDRESS_OFFSET + this.#const_declarations_stack.length + 1;
+    const const_address = this.#declareConst(const_name);
 
-    this.#const_declarations_stack.push({ name: const_name, address: const_address });
-
-    logging.verbose(`DECLARE_VALUE_ADDRESS name=${const_name} address=${const_address}`);
     // retrieve the const_address and write the TNGL_FLAGS with uint16_t variable address value.
-    this.#tnglWriter.writeFlag(TNGL_FLAGS.DECLARE_VALUE_ADDRESS);
+    this.#tnglWriter.writeFlag(TNGL_FLAGS.DECLARE_VARIABLE);
     this.#tnglWriter.writeUint16(const_address);
```

Biome (1.9.4)

[error] This code is unreachable

... because this statement will throw an exception beforehand


(lint/correctness/noUnreachable)